gapis/database/database.go

```diff
@@ -54,6 +59,26 @@ func Build(ctx context.Context, r Resolvable) (interface{}, error) {
 	return Get(ctx).Resolve(ctx, id)
 }
 
+// GetBuilt stores resolvable into d, if the object is already resolved
+// in the database, returns it. Otherwise it returns nil.
+func GetBuilt(ctx context.Context, r Resolvable) (interface{}, error) {
+	id, err := Store(ctx, r)
+	if err != nil {
+		return nil, err
+	}
+	if Get(ctx).IsResolved(ctx, id) {
+		return Get(ctx).Resolve(ctx, id)
+	}
+
+	newCtx := status.PutTask(keys.Clone(context.Background(), ctx), nil)
+	go func() {
+		cctx := status.PutTask(newCtx, nil)
+		Get(cctx).Resolve(cctx, id)
+	}()
```

Comment:
Not quite sure if this background resolving is good to be here. This means this function `GetBuilt()` has side effect not reflect in its name and blurbs. But is it really necessary? Shouldn't we always rely on pre-resolved dependency graph (the only use of this function is trying to get resolved dependency graph)?

Reply:
So in this case, we want to use the dependency graph if it is ready, but also NOT use it if it is not ready yet. This way we don't have to block un-necessarily.

It might be better to update the name of the function, 
GetOrBuild() maybe?

Reply:
GetOrBuild() sounds a good idea.